src/js/actions/tools.js

```diff
@@ -601,7 +603,32 @@ define(function (require, exports) {
                 .then(function (policyID) {
                     this.dispatch(events.tool.VECTOR_MASK_POLICY_CHANGE, policyID);
                 });
+            removePromise = Promise.resolve();
         } else {
+            // delete empty masks when leaving mask mode
+            removePromise = this.transfer(layers.resetLayers, currentDocument, currentLayer)
+                .bind(this)
+                .then(function () {
+                    currentLayer = appStore.getCurrentDocument().layers.selected.first();
```

Comment:
are you protected from multiple layers being selected?  Probably somewhere... but still, this seems a bit dangerous.  File it away for future reference.